pac4j-saml/src/main/java/org/pac4j/saml/metadata/SAML2ServiceProviderMetadataResolver.java

```diff
@@ -126,7 +127,11 @@ public final MetadataResolver resolve() {
                     logger.info("Metadata file already exists at {}.", this.spMetadataResource.getFilename());
                 } else {
                     logger.info("Writing sp metadata to {}", this.spMetadataResource.getFilename());
-
+                    final File parent = spMetadataResource.getFile().getParentFile();
+                    if (!parent.mkdirs()) {
+                        logger.error("Could not construct the directory structure for SP metadata {}", 
```

Comment:
Shouldn't we throw a `TechnicalException` here as we were unable to create the directory where the SP metadata will be written?